### DISCUSSION TITLE
Double dash to disable parsing

### DISCUSSION THREAD

**Discussion by amgcc** [pallets/click#2596] [Q&A]
I need to accept some arguments that should be ignored by click.  I am aware that it is possible to enable extra arguments however it has some major drawbacks.  Use of this feature means that misspelled arguments are ignored which can cause issues.  I think it would be helpful to be able to run a click command with a mix of parsed and unparsed arguments.  In practice it would work something like this:

```
command --parsed-arg1=x --parsed-arg2=y -- --unparsed-args ...
```

If --parsed-arg1 was misspelled in this case it would be flagged as an error.  Any arguments after the double-dash would be ignored by click.

If this proposal is rejected, maybe there is a way to hook into post parsing and add custom validation on the extra arguments for unexpected contents.

**Comment by davidism** [comment]
This already exists.

**Comment by amgcc** [comment]
@davidism Please elaborate.  Is this documented anywhere?  I found no references to double-dashes in the click documentation.

**Reply by tachyondecay** [reply]
https://click.palletsprojects.com/en/8.1.x/arguments/#option-like-arguments

**Reply by amgcc** [reply]
Thank you @tachyondecay.  Search engines won't resolve the terms "click double dash" unless those words are listed in the documentation.  Might be helpful to update accordingly so that others can readily find this information.  At least this thread might get indexed for the next person.

**Comment by ashb** [comment]
Edit: never mind, this is caused by `poetry run` stripping out the `--` param.

This seems broken on 8.1.6, that or it doesn't behave how I expected:

```
❯ poetry run python ./foo.py -- -foo.txt bar.txt
8.1.6
Usage: foo.py [OPTIONS] [FILES]...
Try 'foo.py --help' for help.

Error: No such option: -f
```

```python
import click

print(click.__version__)

@click.command()
@click.argument('files', nargs=-1, type=click.Path())
def touch(files):
    """Print all FILES file names."""
    for filename in files:
        click.echo(filename)

if __name__ == "__main__":
    touch()
```

**Comment by maartenbreddels** [comment]
FYI, https://github.com/pallets/click/issues/619 is the first hit I found on google, but it's locked so it cannot be commented on, and it will also not backlink if you link to it, making this answer hard to find. 
Another solution I found is to explicitly pass the args, and split off the part after `--` yourself. See https://github.com/widgetti/solara/pull/558
```python
    args = sys.argv
    # skip everything after -- if it exists
    if "--" in args:
        args = args[: args.index("--")]
    cli(args[1:])
```
